server/controllers/userController.js

```diff
@@ -124,35 +151,42 @@ class UserController {
       });
   }
   /**
-   * 
+   * Method used to create new user
+   * @param{Object} request - Server Request
+   * @param{Object} response - Server Response
+   * @returns{Void} return Void
    */
   static loginUser(request, response) {
-    Users.findOne({where:{email: request.body.email}})
-      .then(user => {
-         if(user && user.validPassword(request.body.password)){
+    Users.findOne({ where: { email: request.body.email } })
```

Comment:
This login is fine as it is but it has just one little flaw. It can be hacked with what is called the time attack. Here, you are trying to find the user and if the user is not found, you return an error. This returns faster than when a user is found and you try to decode the password. A hacker will know when he has guessed an email correctly and when he has not. A work around this is to create a new instance of a user without a password if the user is not found and ALWAYS check for the password. In this case, it will take about the same time for when a correct user is found or not.

```JavaScript
User.findOne({ where: { email: req.body.email } })
      .then((user) => {
         const user = user || // make a new user instance but don't save it to the database.
            const token = jwt.sign({
            RoleId: user.RoleId,
      	    UserId: user.id
         }, SECRET_KEY, { expiresIn: 86400 });
 -          response.status(201).send({token, expiresIn: 86400})
           }else{
               res.status(401)
              .send({ message: 'Invalid Login Details!' });
         }
       });
   },
```
This might not be totally correct, but I think you should get the idea.